packages/yew-macro/tests/html_macro/component-fail.stderr

```diff
@@ -10,229 +10,311 @@ error: unexpected end of input, expected identifier
 44 |     html! { <Child:: /> };
    |             ^^^^^^^^^^^
 
-error: expected expression following this `with`
+error: `with` doesn't have a value. (hint: set the value to `true` or `false` for boolean attributes)
   --> $DIR/component-fail.rs:45:20
    |
 45 |     html! { <Child with /> };
    |                    ^^^^
 
+error: unexpected end of input, expected expression
+  --> $DIR/component-fail.rs:46:13
+   |
+46 |     html! { <Child .. /> };
+   |             ^^^^^^^^^^^^
+
 error: `props` doesn't have a value. (hint: set the value to `true` or `false` for boolean attributes)
-  --> $DIR/component-fail.rs:46:20
+  --> $DIR/component-fail.rs:47:20
    |
-46 |     html! { <Child props /> };
+47 |     html! { <Child props /> };
    |                    ^^^^^
 
+error: `with` doesn't have a value. (hint: set the value to `true` or `false` for boolean attributes)
+  --> $DIR/component-fail.rs:48:20
+   |
+48 |     html! { <Child with props > };
+   |                    ^^^^
+
 error: this opening tag has no corresponding closing tag
-  --> $DIR/component-fail.rs:47:13
+  --> $DIR/component-fail.rs:49:13
    |
-47 |     html! { <Child with props > };
-   |             ^^^^^^^^^^^^^^^^^^^
+49 |     html! { <Child ..props > };
+   |             ^^^^^^^^^^^^^^^^
 
-error: there are two `with <props>` definitions for this component (note: you can only define `with <props>` once)
-  --> $DIR/component-fail.rs:49:20
+error: `with` doesn't have a value. (hint: set the value to `true` or `false` for boolean attributes)
+  --> $DIR/component-fail.rs:51:20
    |
-49 |     html! { <Child with p1 with p2 /> };
-   |                    ^^^^^^^
+51 |     html! { <Child with p1 with p2 /> };
+   |                    ^^^^
 
-error: `ref` can only be set once
-  --> $DIR/component-fail.rs:50:40
+error: `with` doesn't have a value. (hint: set the value to `true` or `false` for boolean attributes)
+  --> $DIR/component-fail.rs:53:20
    |
-50 |     html! { <Child with props ref={()} ref={()} /> };
-   |                                        ^^^
+53 |     html! { <Child with props ref={()} ref={()} /> };
+   |                    ^^^^
 
-error: `ref` can only be set once
-  --> $DIR/component-fail.rs:51:40
+error: base props expression must appear last in list of props
+  --> $DIR/component-fail.rs:54:28
    |
-51 |     html! { <Child with props ref={()} ref={()} value=1 /> };
-   |                                        ^^^
+54 |     html! { <Child ..props ref={()} ref={()} /> };
+   |                            ^^^
```

Comment:
I think the span here should be on `..props` and not `ref`. 
When parsing `ComponentProps` I think you could move things around to achieve this with `new_spanned`.